### ISSUE TITLE
webrtc-ios. hungUpByUser doesn't work

### ISSUE THREAD

**Issue by neshyurik** [QuickBlox/quickblox-ios-sdk#733]
When I call` [self.session hangUp:@{@"hangup" : @"goBack"}];`  on the first device, delegate's method `- (void)session:(QBRTCSession *)session hungUpByUser:(NSNumber *)userID userInfo:(NSDictionary<NSString *,NSString *> *)userInfo` doesn't call on the second one.

**Comment by Raikerian** [comment]
Hi,

Please check whether you are subscribed to QBRTCClientDelegate protocol.
Also please provide logs for both sides.

**Comment by neshyurik** [comment]
I've subscribed to QBRTCClientDelegate protocol and other methods are called correctly. 
Logs for first side: 
```
  2017-01-25 17:14:05.252 rtc::[SESS]<0E0C74E7-1E85-4F4C-AB71-7742550F9EE1, I:21905758, O:[23198210], T:V> Hang Up call
2017-01-25 17:14:05.253 rtc::[SESS]<0E0C74E7-1E85-4F4C-AB71-7742550F9EE1, I:21905758, O:[23198210], T:V> Begin hangup, userinfo: {
    hangup = goBack;
}.
2017-01-25 17:14:05.254 rtc::[PCON]<21905758, R:Answer> Closing
2017-01-25 17:14:05.289 rtc::[PCON]<21905758, R:Answer> Did change ice connection state to CLOSED
2017-01-25 17:14:05.290 rtc::[PCON]<21905758, R:Answer> Signaling state CLOSED
2017-01-25 17:14:05.883 rtc::dealloc QBRTCAudioTrack
2017-01-25 17:14:05.884 rtc::dealloc QBRTCVideoTrack
2017-01-25 17:14:05.884 rtc::Deallocating local media stream.
2017-01-25 17:14:05.884 remotecam[1108:233488] [ChatService] Message SNT: <message id="24067708" type="headline" to="21905758-28227@chat.quickblox.com"><extraParams xmlns="jabber:client"><callType>1</callType><callerID>21905758</callerID><moduleIdentifier>WebRTCVideoChat</moduleIdentifier><platform>ios</platform><sdkVersion>2.3.1</sdkVersion><sessionID>0E0C74E7-1E85-4F4C-AB71-7742550F9EE1</sessionID><signalType>hangUp</signalType><userInfo><hangup>goBack</hangup></userInfo><opponentsIDs><opponentID>23198210</opponentID></opponentsIDs></extraParams></message>
2017-01-25 17:14:05.885 rtc::QBRTCCameraCapture did remove from video track.
2017-01-25 17:14:05.885 rtc::dealloc QBRTCLocalVideoTrack
2017-01-25 17:14:05.885 rtc::dealloc QBRTCLocalAudioTrack
2017-01-25 17:14:05.887 rtc::[VSND] Stop
```
Logs for second side: 
```
2017-01-25 17:20:14.995 rtc::[SESS]<394C4B02-5474-4D70-8777-795349DF9A71, I:21905758, O:[23198210], T:V> Hang Up call
2017-01-25 17:20:14.996 rtc::[SESS]<394C4B02-5474-4D70-8777-795349DF9A71, I:21905758, O:[23198210], T:V> Begin hangup, userinfo: {
    hangup = "hang up";
}.
2017-01-25 17:20:14.996 rtc::[PCON]<23198210, R:Offer> Closing
2017-01-25 17:20:14.998 rtc::[PCON]<23198210, R:Offer> Did change ice connection state to CLOSED
2017-01-25 17:20:14.998 rtc::[PCON]<23198210, R:Offer> Signaling state CLOSED
2017-01-25 17:20:15.025 rtc::dealloc QBRTCAudioTrack
2017-01-25 17:20:15.025 rtc::dealloc QBRTCVideoTrack
2017-01-25 17:20:15.025 rtc::Deallocating local media stream.
2017-01-25 17:20:15.026 rtc::dealloc QBRTCLocalVideoTrack
2017-01-25 17:20:15.026 rtc::dealloc QBRTCLocalAudioTrack
2017-01-25 17:20:15.026 rtc::[VSND] Stop
2017-01-25 17:20:15.095 remotecam[15027:221364] ConnectionClosedForUser
2017-01-25 17:20:15.096 remotecam[15027:221647] [ChatService] Message SNT: <message id="205044122" type="headline" to="23198210-28227@chat.quickblox.com"><extraParams xmlns="jabber:client"><callType>1</callType><callerID>21905758</callerID><moduleIdentifier>WebRTCVideoChat</moduleIdentifier><platform>ios</platform><sdkVersion>2.3.1</sdkVersion><sessionID>394C4B02-5474-4D70-8777-795349DF9A71</sessionID><signalType>hangUp</signalType><userInfo><hangup>hang up</hangup></userInfo><opponentsIDs><opponentID>23198210</opponentID></opponentsIDs></extraParams></message>
```

**Comment by Raikerian** [comment]
We will push the new release tomorrow which will contain this fix, thank you for your report.

**Comment by Raikerian** [comment]
We have recently released QuickbloxWebRTC 2.4 which contains this fix and many more improvements. Full changelog is available [here](https://quickblox.com/developers/Sample-webrtc-ios#Framework_changelog).
Let us know if you will have any other issues by creating a separate thread.

**Comment by Raikerian** [comment]
@vineeshtp answered in your thread. For future issues please restrain from reporting them in old closed ones (referencing old one in the new issue is recommended).
